libcxx/include/__tuple/ignore.h

```diff
@@ -0,0 +1,35 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCPP___TUPLE_IGNORE_H
+#define _LIBCPP___TUPLE_IGNORE_H
+
+#include <__config>
+
+#if !defined(_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER)
+#  pragma GCC system_header
+#endif
+
+#ifndef _LIBCPP_CXX03_LANG
+
+_LIBCPP_BEGIN_NAMESPACE_STD
+
+struct __ignore_type {
+  template <class _Tp>
+  _LIBCPP_HIDE_FROM_ABI constexpr const __ignore_type& operator=(const _Tp&) const noexcept {
+    return *this;
+  }
+};
+
+inline constexpr __ignore_type ignore;
```

Comment:
This is backporting a change in C++17 ([P0607R0](https://wg21.link/p0607r0))... I think in C++11/14 `std::ignore` was supported to have internal linkage (and thus denoted different objects in different TUs).

Reply:
> This is backporting a change in C++17 ([P0607R0](https://wg21.link/p0607r0))... I think in C++11/14 `std::ignore` was supported to have internal linkage (and thus denoted different objects in different TUs).

What's the benefit of doing this?

Reply:
> What's the benefit of doing this?

This is what libc++ is doing currently (since 8643bdd0165c494f86222fe892a7785ef95e1750). I guess it would be better to backport `inline` additions in a concentrated PR if wanted.

Reply:
@frederick-vs-ja Thanks for the review. Isn't the current implementation in STL in line with what's I implemented here also?

Reply:
MSVC STL is currently marking these non-templated `constexpr` variables `_INLINE_VAR`, which expands to `inline` only since C++17. `_INLINE_VAR` expands to nothing in C++14 mode, which makes `constexpr` variables non-inline and have internal linkage. I think libc++ is effectively doing the same.